oar/core/worksheet_mgr.py

```diff
@@ -0,0 +1,311 @@
+import gspread
+import gspread_formatting
+import os
+import logging
+from oar.core.exceptions import WorksheetException
+from oar.core.config_store import ConfigStore
+from oar.core.const import *
+from oar.core.advisory_mgr import AdvisoryManager, Advisory
+from oar.core.jira_mgr import JiraManager
+from google.oauth2.service_account import Credentials
+from gspread.exceptions import *
+from gspread import Worksheet
+
+logger = logging.getLogger(__name__)
+
+
+class WorksheetManager:
+    """
+    WorksheetManager will be used to update test report with info provided by ConfigStore
+    """
+
+    def __init__(self, cs: ConfigStore):
+        if cs:
+            self._cs = cs
+        else:
+            raise WorksheetException("argument config store is required")
+
+        # init gspread instance with scopes an sa file
+        sa_file_path = self._cs.get_google_sa_file()
+        if os.path.isfile(sa_file_path):
+            try:
+                cred = Credentials.from_service_account_file(
+                    sa_file_path,
+                    scopes=[
+                        "https://spreadsheets.google.com/feeds",
+                        "https://www.googleapis.com/auth/drive",
+                    ],
+                )
+            except Exception as ce:
+                raise WorksheetException("init cred with SA file failed") from ce
+        else:
+            raise WorksheetException(f"SA file path is invalid: {sa_file_path}")
+
+        try:
+            self._gs = gspread.authorize(cred)
+        except Exception as ge:
+            raise WorksheetException("gspread auth failed") from ge
+
+    def create_test_report(self):
+        """
+        Create new report sheet from template
+        Update test report with info in ConfigStore
+        """
+        try:
+            # duplicate template as new test report
+            self._doc = self._gs.open_by_key(self._cs.get_report_template())
+            # check template worksheet exists or not
+            try:
+                template = self._doc.worksheet("template")
+            except WorksheetNotFound as we:
+                raise WorksheetException("cannot find template worksheet") from we
+
+            # check report worksheet exists or not, if yes, skip duplicating
+            try:
+                existing_sheet = self._doc.worksheet(self._cs.release)
+                self._report = TestReport(existing_sheet, self._cs)
+            except WorksheetNotFound:
```

Comment:
@rioliu-rh how about if it is not WorksheetNotFound? we should raise exception.

Reply:
the logic is check target worksheet exists or not, if yes, don't duplicate from template, just update it with build data

Reply:
if other exception is raised, the code does not catch it.